controllers/argocd/status.go

```diff
@@ -212,3 +221,70 @@ func (r *ReconcileArgoCD) reconcileStatusServer(cr *argoprojv1a1.ArgoCD) error {
 	}
 	return nil
 }
+
+// reconcileStatusHost will ensure that the host status is updated for the given ArgoCD.
+func (r *ReconcileArgoCD) reconcileStatusHost(cr *argoprojv1a1.ArgoCD) error {
+	cr.Status.Host = ""
+	cr.Status.Phase = "Available"
+	if cr.Spec.Server.Route.Enabled {
+		if !IsRouteAPIAvailable() {
+			log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
+			return nil
+		}
+		route := newRouteWithSuffix("server", cr)
+		if !argoutil.IsObjectFound(r.Client, cr.Namespace, route.Name, route) {
+			log.Info("argocd-server route requested but not found on cluster")
+			return nil
+		} else {
+			// status.ingress not available
+			if route.Status.Ingress == nil {
+				cr.Status.Host = ""
+				cr.Status.Phase = "Pending"
+			} else {
+				// conditions exist and type is RouteAdmitted
+				if len(route.Status.Ingress[0].Conditions) > 0 && route.Status.Ingress[0].Conditions[0].Type == routev1.RouteAdmitted {
+					if route.Status.Ingress[0].Conditions[0].Status == corev1.ConditionTrue {
+						cr.Status.Host = route.Status.Ingress[0].Host
+						cr.Status.Phase = "Available"
+					} else {
+						cr.Status.Host = ""
+						cr.Status.Phase = "Pending"
+					}
+				} else {
+					// no conditions are available
+					if route.Status.Ingress[0].Host != "" {
+						cr.Status.Host = route.Status.Ingress[0].Host
+						cr.Status.Phase = "Available"
+					} else {
+						cr.Status.Host = "Unavailable"
+						cr.Status.Phase = "Pending"
+					}
+				}
+			}
+		}
+	} else if cr.Spec.Server.Ingress.Enabled {
+		ingress := newIngressWithSuffix("server", cr)
+		if !argoutil.IsObjectFound(r.Client, cr.Namespace, ingress.Name, ingress) {
+			log.Info("argocd-server ingress requested but not found on cluster")
+			return nil
+		} else {
+			if !reflect.DeepEqual(ingress.Status.LoadBalancer, corev1.LoadBalancerStatus{}) && len(ingress.Status.LoadBalancer.Ingress) > 0 {
```

Comment:
@reginapizza @jannfis I can''t figure out why is the DeepEqual() is needed.

Reply:
would `if !reflect.ValueOf(ingress.Status.LoadBalancer).IsZero() && len(ingress.Status.LoadBalancer.Ingress) > 0 {` be better?

Reply:
Yes, `IsZero` a bit clean and faster then `DeepCopy` and `&& len(ingress.Status.LoadBalancer.Ingress) > 0` can be eliminated.

Reply:
I think we should keep the `len(ingress.Status.LoadBalancer.Ingress) > 0` as a sanity check, to make sure code only executes if we have at least 1 proper ingress hostname/ip to work with